File: deque/deque.js
function Deque() {
    this.deque_list = [];
}

Deque.prototype.addFront = function(data) {
    this.deque_list.push(data);
}

Deque.prototype.addRear = function(data) {
    this.deque_list.splice(0,0,data);
}

Deque.prototype.removeFront = function() {
    return this.deque_list.pop();
}

Deque.prototype.removeRear = function() {
    return this.deque_list.splice(0,1)[0];
}

Deque.prototype.isEmpty = function() {
    return this.deque_list.length === 0;
}

Deque.prototype.size = function() {
    return this.deque_list.length;
}

//TESTING FUNCTIONS

function print(data) {
    console.log(data);
}

print("Initializing Deque");
dq = new Deque();
print("AddFront: 'bar'");
dq.addFront("bar");
print("AddRear: 'foo'");
dq.addRear("foo");
print("AddFront: 'Hello'");
dq.addFront("Hello");
print("AddRear: 'World'");
dq.addRear("World");
print( "Size: " + dq.size() );
print("Removing Front +  Rear");
print( dq.removeFront() + ' ' + dq.removeRear() );
print("Removing Rear + Front");
print( dq.removeRear() + ' ' + dq.removeFront() );
print( "isEmpty: " + dq.isEmpty() );

File: deque/deque.py
class Deque:
    def __init__(self):
        self.deque_list = []

    def addFront(self, data):
        self.deque_list.append(data)

    def addRear(self, data):
        self.deque_list.insert(0, data)

    def removeFront(self):
        return self.deque_list.pop()

    def removeRear(self):
        return self.deque_list.pop(0)

    def isEmpty(self):
        return self.deque_list == []

    def size(self):
        return len(self.deque_list)